utils/build-script-impl

```diff
@@ -1001,138 +998,245 @@ if [[ ! "${SKIP_BUILD_SWIFTPM}" ]] ; then
+# - "LOCAL_HOST" is Build (local machine running this script)
+# - "CROSS_COMPILE_HOSTS" are the Hosts (implicitly includes LOCAL_HOST)
+# - "STDLIB_DEPLOYMENT_TARGETS" are the Targets
+
+CROSS_COMPILE_HOSTS=($CROSS_COMPILE_HOSTS)
+for t in ${CROSS_COMPILE_HOSTS} ; do
+    case ${t} in
+        iphone* | appletv* | watchos* )
             ;;
         *)
-            echo "Unknown compiler deployment target: ${deployment_target}"
+            echo "Unknown deployment target: ${t}"
             exit 1
             ;;
     esac
-    if [[ "${build_for_this_target}" ]] ; then
-        if [[ "${BUILD_SWIFT_STDLIB_UNITTEST_EXTRA}" == "1" ]] ; then
-            SWIFT_STDLIB_TARGETS+=("swift-stdlib-${deployment_target}")
-        else
-            if [[ "${VALIDATION_TEST}" == "1" || "${LONG_TEST}" == "1" ]] ; then
-                SWIFT_STDLIB_TARGETS+=("swift-stdlib-${deployment_target}")
-            else
-                SWIFT_STDLIB_TARGETS+=("swift-test-stdlib-${deployment_target}")
-            fi
+done
+
+ALL_HOSTS=("${LOCAL_HOST}" "${CROSS_COMPILE_HOSTS[@]}")
+
+function is_cross_tools_host() {
+    local host="$1"
+    for t in "${CROSS_COMPILE_HOSTS[@]}" ; do
+        if [ "${host}" == "${t}" ] ; then
+            echo 1
         fi
+    done
+}
+
+# We only install in to host-specific subdirectories if there are cross-compile
+# hosts defined. Other users will expect their products at INSTALL_DESTDIR.
+function has_cross_compile_hosts() {
+    if [[ ${#ALL_HOSTS[@]} -gt 1 ]]; then
+        echo "1"
     fi
-    if [[ "${build_benchmark_this_target}" ]] ; then
-        SWIFT_BENCHMARK_TARGETS+=("swift-benchmark-${deployment_target}")
-        if [[ $(not ${SKIP_TEST_BENCHMARK}) ]] ; then
-          SWIFT_RUN_BENCHMARK_TARGETS+=("check-swift-benchmark-${deployment_target}")
-        fi
+}
+
+# When building cross-compilers for these hosts, merge all of their contents together with lipo
+function should_include_host_in_lipo() {
+    local host="$1"
+    if [[ -z "${SKIP_MERGE_LIPO_CROSS_COMPILE_TOOLS}" ]]; then
+        case ${host} in 
+            iphone* | appletv* | watchos* )
+                echo 1
+                ;;
+        esac
     fi
-    if [[ "${test_this_target}" ]] ; then
-        test_target_suffix=""
-        if [[ -n "${test_host_only}" ]] ; then
-            test_target_suffix="-non-executable"
-        fi
+}
+
+function host_has_darwin_symbols() {
+    local host="$1"
+    case ${host} in 
+        macosx* | iphone* | appletv* | watchos* )
+            echo 1
+            ;;
+    esac
+}
 
-        test_subset_target_suffix=""
-        if [[ "${VALIDATION_TEST}" == "1" ]] ; then
-            if [[ "${LONG_TEST}" == "1" ]] ; then
-                test_subset_target_suffix="-all"
+# FIXME: We currently do not support cross-compiling swift with compiler-rt.
+if [[ "${CROSS_COMPILE_HOSTS}" ]]; then
+    SKIP_COMPILER_RT=1
+fi
+
+STDLIB_DEPLOYMENT_TARGETS_DEFAULT="${STDLIB_DEPLOYMENT_TARGETS}"
```

Comment:
Where does `STDLIB_DEPLOYMENT_TARGETS` get assigned? Also, why do we need to copy it to `STDLIB_DEPLOYMENT_TARGETS_DEFAULT`?